src/CommunityToolkit/Xamarin.CommunityToolkit/ObjectModel/Extensions/INotifyPropertyChangedEx.shared.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.ComponentModel;
+
+namespace Xamarin.CommunityToolkit.ObjectModel.Extensions
+{
+	public static class INotifyPropertyChangedEx
+	{
+		public static void WeakSubscribe<T>(this INotifyPropertyChanged target, T subscriber, Action<T, object, EventArgs> action)
+		{
+			var weakSubscriber = new WeakReference(subscriber, false);
+			target.PropertyChanged += handler;
+
+			void handler(object sender, PropertyChangedEventArgs e)
+			{
+				var s = (T)weakSubscriber.Target;
+				if (s is null)
```

Comment:
bear with, but we don't use pattern matching to check for null.
```suggestion
				if (s == null)
```

Reply:
Yeah, forgot about it. Do you now why this is the rule here? `is` doesn't call any equality overloads, so it always returns `true` if reference type has no value.

Reply:
Yeap, I agree with you that `is null` and `is { }` ( != null) is better to check nulls. I'm not sure about it but I believe that's because of the Xamarin pattern. All Xamarin projects (Forms, Essentials, etc) use the `==` and `!=` pattern.

Reply:
Got it. BTW as of C# 9 you can use `is not null`